### DISCUSSION TITLE
'window.location.reload' vs 'router.reload'

### DISCUSSION THREAD

**Discussion by darelover** [vercel/next.js#53517] [Help] [answered]
### Summary

Does 'router.reload' have any benefit over 'window.location.reload' ?

Related: https://github.com/vercel/next.js/discussions/38400

**Comment by icyJoseph** [comment]
If you do `replace` on the Router from Next.js, you are letting the Next.js Router logic, "know" about the replace change, so that it doesn't have to trigger a page unload + reload, but instead it first tries to change what components are shown based on the new Router state, as part of this process the Next Router also updates the history object. 


When you call `location.replace` you are directly telling the browser to mutate the history object and cause a page unload + reload on to the replaced page.

**Reply by darelover** [reply]
I miswrote `reload` as `replace`. I have updated the description now

**Reply by icyJoseph** [reply, marked as answer]
> Reload the current URL. Equivalent to clicking the browser’s refresh button. It executes window.location.reload().


There's no real difference. In the App Router they didn't even include that method on the router. 


What's your concern? there's no real benefit, other than maybe a bit shorter code expression.

**Reply by darelover** [reply]
No concern really. They other discussion that I linked to made me think if there was any benefit. Thanks a lot

**Reply by icyJoseph** [reply]
I see, nah, you are good, just for completeness sake:

- https://github.com/vercel/next.js/blob/canary/packages/next/src/shared/lib/router/router.ts#L1007-L1009

This is the code behind `router.reload`:

```ts
  reload(): void {
    window.location.reload()
  }
```

Don't forget to mark as answered if you are satisfied :)
